docs/webapi/migratingfromwebapi2/migratingfromwebapi2.rst

```diff
@@ -0,0 +1,145 @@
+	:emphasize-lines: 14
+	:linenos:
+
+WebApiConfig is defined in App_Start, and has just one static Register method:
+
+.. literalinclude:: ../../../samples/WebAPIMigration/ProductsApp/App_Start/WebApiConfig.cs
+	:language: c#
+	:emphasize-lines: 15-20
+	:linenos:
+
+This class configures `attribute routing <http://www.asp.net/web-api/overview/web-api-routing-and-actions/attribute-routing-in-web-api-2>`_, which isn't actually being used in this project, as well as the routing table that Web API 2 uses. In this case, Web API will expect URLs to match /api/{controller}/{id}, with {id} being optional.
+
+The ProductsApp project includes just one simple controller, which inherits from ApiController and exposes two methods:
+
+.. literalinclude:: ../../../samples/WebAPIMigration/ProductsApp/Controllers/ProductsController.cs
+	:language: c#
+	:emphasize-lines: 19,24
+	:linenos:
+
+Finally, the model, Product, used by the ProductsApp, is a simple class:
+
+.. literalinclude:: ../../../samples/WebAPIMigration/ProductsApp/Models/Product.cs
+	:language: c#
+	:linenos:
+
+Create the Destination Project
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Using Visual Studio 2015, create a new, empty solution, and add the existing ProductsApp project to it. Then, add a new Web Project to the solution. In this case, I'm naming the new project 'ProductsDnx'.
+
+.. image:: _static/add-web-project.png
+
+Next, choose the ASP.NET 5 Web API template project. We will migrate the ProductsApp contents to this new project.
+
+.. image:: _static/aspnet-5-webapi.png
+
+Delete the Project_Readme.html file from the new project. You can also delete the Your solution should now look like this:
+
+.. image:: _static/webapimigration-solution.png
+
+Migrate Configuration
+^^^^^^^^^^^^^^^^^^^^^
+
+ASP.NET 5 no longer uses global.asax, web.config, or App_Start folders. Instead, all startup tasks are done in Startup.cs in the root of the project, and static configuration files can be wired up from there if needed. Since Web API is now built into ASP.NET 5, there is less need to configure it. Attribute-based routing is now included by default when UseMvc() is called, and this is how the Web API starter project handles routing.
+
+.. literalinclude:: ../../../samples/WebAPIMigration/ProductsDnx/Startup.cs
+	:language: c#
+	:emphasize-lines: 27
+	:linenos:
+	
+If you want to use attribute routing in your project going forward, you don't need to do any additional configuration. You can simply apply the attributes as needed to the API methods or controllers, as is done in the sample ValuesController.cs class that is included in the Web API starter project:
+
+.. literalinclude:: ../../../samples/WebAPIMigration/ProductsDnx/Controllers/ValuesController.cs
+	:language: c#
+	:emphasize-lines: 8,12,19,32,38
+	:linenos:
+
+In this case, once we copy ProductsController to the new project, we would simply include the route attribute on the controller:
+
+.. code-block:: c#
+
+	[Route("api/[controller]")]
```

Comment:
The route token syntax (here used for inserting the controller name in the route) is new and is worth explaining or pointing to a more general routing document with more details